src/get-peer-id.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const PeerId = require('peer-id')
+const multiaddr = require('multiaddr')
+const errCode = require('err-code')
+
+const { codes } = require('./errors')
+
+/**
+ * Converts the given `peer` to a `PeerId` instance.
+ * If a multiaddr is received, the addressBook is updated.
+ * @param {PeerId|Multiaddr|string} peer
+ * @param {PeerStore} peerStore
+ * @returns {PeerId}
+ */
+function getPeerId (peer, peerStore) {
+  if (typeof peer === 'string') {
+    peer = multiaddr(peer)
+  }
+
+  let addr
+  if (multiaddr.isMultiaddr(peer)) {
+    addr = peer
+    try {
+      peer = PeerId.createFromB58String(peer.getPeerId())
+    } catch (err) {
+      throw errCode(
+        new Error(`${peer} is not a valid peer type`),
+        codes.ERR_INVALID_MULTIADDR
+      )
+    }
+  }
+
+  if (addr && peerStore) {
+    peerStore.addressBook.add(peer, [addr])
```

Comment:
Since we're redoing this function it might be a good time to remove the update from it as it's a bit error prone (doing a get that includes a write). Perhaps this should be `getPeer` and we return `{ id, multiaddrs }`? Then the calling function can choose whether or not it wants to store the address.